go/store/prolly/tree/json_diff_test.go

```diff
@@ -183,43 +249,164 @@ var simpleJsonDiffTests = []jsonDiffTest{
 	},
 	{
 		name: "modifications returned in lexographic order",
-		from: types.JsonObject{"a": types.JsonObject{"1": "i"}, "aa": 2, "b": 6},
-		to:   types.JsonObject{"": 1, "a": types.JsonObject{}, "aa": 3, "bb": 5},
+		from: types.JSONDocument{types.JsonObject{"a": types.JsonObject{"1": "i"}, "aa": 2, "b": 6}},
+		to:   types.JSONDocument{types.JsonObject{"": 1, "a": types.JsonObject{}, "aa": 3, "bb": 5}},
 		expectedDiffs: []JsonDiff{
 			{
-				Key:  "$.\"\"",
+				Key:  makeJsonPathKey(``),
 				To:   &types.JSONDocument{Val: 1},
 				Type: AddedDiff,
 			},
 			{
-				Key:  "$.\"a\".\"1\"",
+				Key:  makeJsonPathKey(`a`, `1`),
 				From: &types.JSONDocument{Val: "i"},
 				Type: RemovedDiff,
 			},
 			{
-				Key:  "$.\"aa\"",
+				Key:  makeJsonPathKey(`aa`),
 				From: &types.JSONDocument{Val: 2},
 				To:   &types.JSONDocument{Val: 3},
 				Type: ModifiedDiff,
 			},
 			{
-				Key:  "$.\"b\"",
+				Key:  makeJsonPathKey(`b`),
 				From: &types.JSONDocument{Val: 6},
 				Type: RemovedDiff,
 			},
 			{
-				Key:  "$.\"bb\"",
+				Key:  makeJsonPathKey(`bb`),
 				To:   &types.JSONDocument{Val: 5},
 				Type: AddedDiff,
 			},
 		},
 	},
 }
 
+func makeLargeJsonDiffTests(t *testing.T) []jsonDiffTest {
+	ctx := sql.NewEmptyContext()
+	ns := NewTestNodeStore()
+
+	_ = func(document types.MutableJSON, path string, val interface{}) types.MutableJSON {
+		jsonVal, inRange, err := types.JSON.Convert(val)
+		require.NoError(t, err)
+		require.True(t, (bool)(inRange))
+		newDoc, changed, err := document.Insert(ctx, path, jsonVal.(sql.JSONWrapper))
+		require.NoError(t, err)
+		require.True(t, changed)
+		return newDoc
+	}
+
+	set := func(document types.MutableJSON, path string, val interface{}) types.MutableJSON {
+		jsonVal, inRange, err := types.JSON.Convert(val)
+		require.NoError(t, err)
+		require.True(t, (bool)(inRange))
+		newDoc, changed, err := document.Replace(ctx, path, jsonVal.(sql.JSONWrapper))
+		require.NoError(t, err)
+		require.True(t, changed)
+		return newDoc
+	}
+
+	lookup := func(document types.SearchableJSON, path string) sql.JSONWrapper {
+		newDoc, err := document.Lookup(ctx, path)
+		require.NoError(t, err)
+		return newDoc
+	}
+
+	_ = func(document types.MutableJSON, path string) types.MutableJSON {
+		newDoc, changed, err := document.Remove(ctx, path)
+		require.True(t, changed)
+		require.NoError(t, err)
+		return newDoc
+	}
+
+	largeObject := createLargeDocumentForTesting2(t, ctx, ns)
+	return []jsonDiffTest{
```

Comment:
How about adding some coverage for added and removed diffs?

Reply:
Good idea. These tests aren't needed for test coverage: we get all that from the added three way merge tests. But I added these tests to help debug some issues that came up while I was writing this, and smaller-scoped tests that test the two-way differ directly are useful. I'll add a couple.